crates/ra_ide_db/src/defs.rs

```diff
@@ -82,38 +82,40 @@ pub enum NameClass {
     Definition(Definition),
     /// `None` in `if let None = Some(82) {}`
     ConstReference(Definition),
+    FieldShorthand {
+        local: Local,
+        field: Definition,
+    },
 }
 
 impl NameClass {
     pub fn into_definition(self) -> Option<Definition> {
         match self {
             NameClass::Definition(it) => Some(it),
             NameClass::ConstReference(_) => None,
+            NameClass::FieldShorthand { local: _, field } => Some(field),
```

Comment:
This should return `local`, as that's the definition which is actually defined by the local. 

```suggestion
            NameClass::FieldShorthand { local, field: _ } => Some(local.into()),
```